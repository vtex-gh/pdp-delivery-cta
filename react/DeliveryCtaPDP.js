import DeliveryCtaPDP from './components/DeliveryCtaPDP/index'

export default DeliveryCtaPDP
