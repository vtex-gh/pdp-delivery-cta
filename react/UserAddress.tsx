import React, { useEffect } from 'react'
import { useQuery } from 'react-apollo'
import { useCssHandles } from 'vtex.css-handles'
import type { CssHandlesTypes } from 'vtex.css-handles'
import ADDRESS_QUERY from 'vtex.store-resources/QueryAddress'

import Container from './Container'
import AddressInfo, {
  CSS_HANDLES as AddressInfoCssHandles,
} from './components/UserAddress/AddressInfo'
import { UserAddressCssHandlesProvider } from './components/UserAddress/UserAddressCssHandles'

export const USER_ADDRESS_CSS_HANDLES = [
  'userAddressContainer',
  ...AddressInfoCssHandles,
] as const

type Props = {
  /** Define if address should be displayed in a single line or in blocks */
  variation: 'inline' | 'bar'
  /** Define if street should be displayed */
  showStreet?: boolean
  /** Define if city and state should be displayed */
  showCityAndState?: boolean
  /** Define if postal code should be displayed */
  showPostalCode?: boolean
  /** Define if prefix should be displayed */
  showPrefix?: boolean
  /** Define if component should render if address is empty */
  showIfEmpty?: boolean
  /** Used to override default CSS handles */
  classes?: CssHandlesTypes.CustomClasses<typeof USER_ADDRESS_CSS_HANDLES>
}

function UserAddress({
  variation,
  showStreet = true,
  showCityAndState = false,
  showPostalCode = false,
  showPrefix = true,
  showIfEmpty = false,
  classes,
}: Props) {
  const { handles, withModifiers } = useCssHandles(USER_ADDRESS_CSS_HANDLES, {
    classes,
  })

  // todo: type the data returned
  const { data: orderFormData, refetch } = useQuery(ADDRESS_QUERY, {
    ssr: false,
  })

  useEffect(() => {
    const handleLocationUpdated = () => refetch()

    window.addEventListener('locationUpdated', handleLocationUpdated)

    return () => {
      window.removeEventListener('locationUpdated', handleLocationUpdated)
    }
  }, [refetch])

  const orderForm = orderFormData?.orderForm

  if (!orderForm || (!orderForm?.shippingData?.address && !showIfEmpty)) {
    return null
  }

  const isInline = variation === 'inline'
  const isInverted = !isInline

  const AddressInfoWithHandles = () => {
    return (
      <UserAddressCssHandlesProvider
        handles={handles}
        withModifiers={withModifiers}
      >
        <AddressInfo
          inverted={isInverted}
          inline={isInline}
          orderForm={orderForm}
          showStreet={showStreet}
          showCityAndState={showCityAndState}
          showPostalCode={showPostalCode}
          showPrefix={showPrefix}
          showIfEmpty={showIfEmpty}
        />
      </UserAddressCssHandlesProvider>
    )
  }

  if (isInline) {
    return (
      <div
        className={`${handles.userAddressContainer} ph5`}
        style={{
          maxWidth: '30rem',
        }}
      >
        <AddressInfoWithHandles />
      </div>
    )
  }

  return (
    <div
      className={`${handles.userAddressContainer} bg-base--inverted c-on-base--inverted flex ph5 pointer pv3 ml3 mr3`}
    >
      <Container className="flex justify-center w-100 left-0">
        <div className="w-100 mw9 flex">
          <AddressInfoWithHandles />
        </div>
      </Container>
    </div>
  )
}

export default UserAddress
