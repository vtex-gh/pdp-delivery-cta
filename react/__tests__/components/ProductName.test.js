import React from 'react'
import { render } from '@vtex/test-tools/react'
import PropTypes from 'prop-types'

import ProductName from '../../ProductName'

describe('<ProductName />', () => {
  const defaultProps = {
    loaderClass: '',
    name: 'ProductTest',
  }

  const context = { culture: { currency: 'USD' } }

  const renderComponent = customProps => {
    const props = {
      ...defaultProps,
      ...customProps,
    }

    return render(<ProductName {...props} />, {
      context,
      childContextTypes: {
        culture: PropTypes.object,
      },
    })
  }

  it('should be mounted', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toBeDefined()
  })

  it('should match the snapshot with only Name', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot with Name and SkuName', () => {
    const { asFragment } = renderComponent({
      skuName: 'ProductSkuName',
      showSku: true,
    })

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot with Name and Brand', () => {
    const { asFragment } = renderComponent({
      brandName: 'ProductBrandName',
      showBrandName: true,
    })

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot with Name and Product Reference', () => {
    const { asFragment } = renderComponent({
      productReference: 'productReferenceTest',
      showProductReference: true,
    })

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot with all options', () => {
    const { asFragment } = renderComponent({
      skuName: 'ProductSkuName',
      showSku: true,
      brandName: 'ProductBrandName',
      showBrandName: true,
      productReference: 'productReferenceTest',
      showProductReference: true,
    })

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot Loader', () => {
    const { asFragment } = renderComponent({ name: undefined })

    expect(asFragment()).toMatchSnapshot()
  })
})
