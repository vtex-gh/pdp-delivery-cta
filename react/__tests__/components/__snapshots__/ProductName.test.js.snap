// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductName /> should match the snapshot Loader 1`] = `
<DocumentFragment>
  <h1
    class="productNameContainer mv0 t-heading-4"
  >
    <span
      class="productBrand "
    >
      Billabong Die Cut Loose Fit Short Sleeve Rashguard - Men's 
    </span>
  </h1>
</DocumentFragment>
`;

exports[`<ProductName /> should match the snapshot with Name and Brand 1`] = `
<DocumentFragment>
  <h1
    class="productNameContainer mv0 t-heading-4"
  >
    <span
      class="productBrand "
    >
      ProductTest - ProductBrandName
    </span>
  </h1>
</DocumentFragment>
`;

exports[`<ProductName /> should match the snapshot with Name and Product Reference 1`] = `
<DocumentFragment>
  <h1
    class="productNameContainer mv0 t-heading-4"
  >
    <span
      class="productBrand "
    >
      ProductTest 
    </span>
    <span
      class="productReference "
    >
      REF: productReferenceTest
    </span>
  </h1>
</DocumentFragment>
`;

exports[`<ProductName /> should match the snapshot with Name and SkuName 1`] = `
<DocumentFragment>
  <h1
    class="productNameContainer mv0 t-heading-4"
  >
    <span
      class="productBrand "
    >
      ProductTest 
    </span>
    <span
      class="productBrand "
    >
      ProductSkuName
    </span>
  </h1>
</DocumentFragment>
`;

exports[`<ProductName /> should match the snapshot with all options 1`] = `
<DocumentFragment>
  <h1
    class="productNameContainer mv0 t-heading-4"
  >
    <span
      class="productBrand "
    >
      ProductTest - ProductBrandName
    </span>
    <span
      class="productBrand "
    >
      ProductSkuName
    </span>
    <span
      class="productReference "
    >
      REF: productReferenceTest
    </span>
  </h1>
</DocumentFragment>
`;

exports[`<ProductName /> should match the snapshot with only Name 1`] = `
<DocumentFragment>
  <h1
    class="productNameContainer mv0 t-heading-4"
  >
    <span
      class="productBrand "
    >
      ProductTest 
    </span>
  </h1>
</DocumentFragment>
`;
